### DISCUSSION TITLE
Regarding converting preprocessing pipeline for android conversion

### DISCUSSION THREAD

**Discussion by aitrics-chris** [open-mmlab/mmpose#1200] [Q&A]
Hi, I downloaded the pretrained model from mmpose and would like to use it on android. It looks like that I need to convert the model to torchscript, but what should I do with the preprocessing pipeline, i.e., test_pipeline, which the pretrained model demands?
Thanks.

**Comment by ly015** [comment]
The preprocessing pipeline is not part of the model and will not be exported together with the model. So I am afraid you may need to reimplement it in your application.

BTW, [MMDeploy](https://github.com/open-mmlab/mmdeploy) is working on supporting torchscript conversion for OpenMMLab models. Please stay tuned for their updates.

**Reply by aitrics-chris** [reply]
Though MMDeploy converts OpenMMLab model, the pretrained model demands pre-proessing. For example, if the pre-trained model was trained with the train_pipeline which includes normalization, the test_pipeline must involve normalization too. Therefore, wouldn't I still have to reimplement test_pipeline though I use MMDeploy? Is there any module that supports test_pipeline conversion? Thanks.
